users/views.py

```diff
@@ -86,7 +101,55 @@ def post(self, request):
                     'message': 'Login successful.',
                     'refresh': str(refresh),
                     'access': str(refresh.access_token),
-                })
-            return Response({"message": "Incorrect email or password.invalid credential"}, status=status.HTTP_400_BAD_REQUEST)
+                }, status=status.HTTP_200_OK)
+            return Response({"message": "Incorrect email or password.Please check your email or password and try again."}, status=status.HTTP_400_BAD_REQUEST)
         except User.DoesNotExist:
             return Response({"message": "user with this email not found. Please register first."}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ForgotPasswordView(APIView):
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+        serializer = ForgotPasswordSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        email = serializer.validated_data["email"]
+        try:
+            user = User.objects.get(email=email)
+        except User.DoesNotExist:
+            return Response({"message": "User with this email does not exist."}, status=status.HTTP_400_BAD_REQUEST)
+
+        token = default_token_generator.make_token(user)
+        uid = urlsafe_base64_encode(force_bytes(user.pk))
+
+        reset_link = f"http://localhost:8000/reset-password/{uid}/{token}/"
+

```

Comment:
http://localhost:8000/reset-password/  
it should be from .env not be hardcoded here